src/resources/2015/Dinseylandfill.ts

```diff
@@ -0,0 +1,316 @@
+    this.questLocation = questLocation;
+  }
+
+  currentQuest(): boolean {
+    return get(this.questStateProperty) !== "unstarted";
+  }
+}
+
+const kioskUrl = "place.php?whichplace=airport_stench&action=airport3_kiosk";
+const maintUrl = "place.php?whichplace=airport_stench&action=airport3_tunnels";
+
+export const quests = [
+  new questData(
+    "lube",
+    1,
+    "Track Maintenance",
+    "Super Luber",
+    "questEStSuperLuber",
+    "",
+    0,
+    false,
+    $item`lube-shoes`,
+    $location`Barf Mountain`
+  ),
+  new questData(
+    "fuel",
+    0,
+    "Electrical Maintenance",
+    "Give Me Fuel",
+    "questEStGiveMeFuel",
+    "",
+    0,
+    false,
+    $item`none`,
+    $location`The Toxic Teacups`
+  ),
+  new questData(
+    "sexism",
+    2,
+    "Sexism Reduction",
+    "Social Justice Adventurer I",
+    "questEStSocialJusticeI",
+    "dinseySocialJusticeIProgress",
+    15,
+    true,
+    $item`none`,
+    $location`Pirates of the Garbage Barges`
+  ),
+  new questData(
+    "racism",
+    3,
+    "Racism Reduction",
+    "Social Justice Adventurer II",
+    "questEStSocialJusticeII",
+    "dinseySocialJusticeIIProgress",
+    15,
+    true,
+    $item`none`,
+    $location`Uncle Gator's Country Fun-Time Liquid Waste Sluice`
+  ),
+  new questData(
+    "fun",
+    4,
+    "Compulsory Fun",
+    "Whistling Zippity-Doo-Dah",
+    "questEStZippityDooDah",
+    "dinseyFunProgress",
+    15,
+    true,
+    $item`Dinsey mascot mask`,
+    $location`The Toxic Teacups`
+  ),
+  new questData(
+    "trash",
+    6,
+    "Waterway Debris Removal",
+    "Teach a Man to Fish Trash",
+    "questEStFishTrash",
+    "dinseyFilthLevel",
+    0,
+    true,
+    $item`trash net`,
+    $location`Pirates of the Garbage Barges`
+  ),
+  new questData(
+    "bear",
+    5,
+    "Bear Removal",
+    "Nasty, Nasty Bears",
+    "questEStNastyBears",
+    "dinseyNastyBearsDefeated",
+    8,
+    false,
+    $item`none`,
+    $location`Uncle Gator's Country Fun-Time Liquid Waste Sluice`
+  ),
+  new questData(
+    "food",
+    7,
+    "Guest Sustenance Assurance",
+    "Will Work With Food",
+    "questEStWorkWithFood",
+    "dinseyTouristsFed",
+    30,
+    false,
+    $item`none`,
+    $location`Barf Mountain`
+  ),
+];
+
+export function disposeGarbage(): void {
+  if (!get("_dinseyGarbageDisposed") && haveItem($item`bag of park garbage`)) {
+    visitUrl(maintUrl);
+    runChoice(6);
+  }
+}
+
+export function hasQuest(): boolean {
+  for (const quest of quests) {
+    if (quest.currentQuest()) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+export function activeQuest(): questData {
+  for (const quest of quests) {
+    if (quest.currentQuest()) {
+      return quest;
+    }
+  }
+
+  return new questData(
+    "",
+    -1,
+    "",
+    "",
+    "",
+    "",
+    -1,
+    false,
+    $item`none`,
+    $location`none`
+  );
+}
+
+export function questComplete(): boolean {
+  const quest = activeQuest();
```

Comment:
This is looping our quest list twice - just check if the active quest is the blank quest. Why are we using a blank quest in the first place? Why not just pass null and handle it.

Reply:
I found this easier than trying to stop it complaining about null if I recall correctly.